RAMP/unpacker.py

```diff
@@ -59,18 +64,23 @@ def _validate_zip_file(zip_file, max_bundle_size_mb):
     exceed a certain threshold.
     """
     infolist = zip_file.infolist()
-    # We're expecting exactly two files.
-    if len(infolist) != 2:
+    # We're expecting to have .so file, module.json and deps dir.
+    all_files_len = len([file for file in infolist])
+    so_files_len = len([file for file in infolist if file.filename.endswith(".so")])
+    expected_files_len = so_files_len + len([file for file in infolist if (file.filename == "module.json" or file.filename.startswith("deps/"))])
+
+    if all_files_len > expected_files_len or so_files_len > 1:
         raise UnpackerPackageError(message="module zip file content invalid",
-                                   reason="module zip file should contains exactly two file",
+                                   reason="module zip file should contains exactly two \
+                                           files and optionally a deps folder",
                                    error_code="invalid_number_of_files")
 
     file_size_limit_bytes = max_bundle_size_mb * 1024 * 1024 if max_bundle_size_mb else DEFAULT_MAX_MODULE_FILE_SIZE
 
     # Check zip content size.
     for zip_info in infolist:
         # Size of the compressed/uncompressed data.
-        if zip_info.file_size > file_size_limit_bytes:
+        if not zip_info.filename.startswith("deps/") and zip_info.file_size > file_size_limit_bytes:
```

Comment:
I disagree with this check.
I believe we should do as the comment states --
make sure that the zip size _and_ its total content is not exceeding the size limit

what this check currently validates is that every file on its own does not exceed the limit, which is not "the intention of the poet"

Reply:
who is the poet ? @guyco-redis ask me to leave the check only for the original files to make sure module + json is not big enough.  deciding how and if deps should be size limited is an open question. lets wait for @MeirShpilraien opinion on the subject

Reply:
Today when we download the deps, is it size limited? If not then why should it be limit when it is inside the zip?

Reply:
@MPalarya ? afair, we don't do it today. thus @guyco-redis said we don't need to check the deps size. what do you think ?